chainerx_cc/chainerx/array_body.cc

```diff
@@ -141,7 +139,10 @@ ReturnType ArrayBody::GetGradImpl(ThisPtr this_ptr, const BackpropId& backprop_i
         return nullptr;
     }
     CHAINERX_ASSERT(*i < this_ptr->grads_.size());
-    return this_ptr->grads_[*i].get();
+    ReturnType grad = this_ptr->grads_[*i].get();
+    CHAINERX_ASSERT(grad != nullptr);
+    CHAINERX_ASSERT(GetKind(this_ptr->dtype_) == DtypeKind::kFloat || !grad->has_value());  // Non-integral arrays cannot have gradients.
```

Comment:
You probably meant the opposite here.
```suggestion
    CHAINERX_ASSERT(GetKind(this_ptr->dtype_) == DtypeKind::kFloat || !grad->has_value());  // Integral arrays cannot have gradients.
```

Reply:
Thanks, fixed.